onnxruntime/core/providers/nnapi/nnapi_builtin/builders/shaper.cc

```diff
@@ -460,6 +466,19 @@ Status Shaper::ResizeUsingOutputSizesImpl(const std::string& input_name,
   return Status::OK();
 }
 
+Status Shaper::PadImpl(const std::string& input_name,
+                       const std::vector<int32_t>& pads,
+                       const std::string& output_name) {
+  Shape padded_shape = shape_map_.at(input_name);
+  const size_t rank = padded_shape.size();
+  ORT_RETURN_IF_NOT(pads.size() == 2 * rank, "Expected 2*rank pad values");
+  for (size_t i = 0; i < rank; ++i) {
```

Comment:
nit: including the actual value in the error message may make it more helpful to the user